.github/workflows/release.yaml

```diff
@@ -0,0 +1,68 @@
+name: release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+permissions:
+  actions: write
+  contents: write
+  pull-requests: write
```

Comment:
Can you educate me on this.  Why are these last 2 required?

Reply:
I still need to mess around with those permissions. The github action to generate a tag needs some permissions to save the tag in the repo.

Reply:
Would we not push the tag ourselves?  What is the workflow you imagine?

Reply:
Sorry, the gh action is used to generate the release when a tag is pushed. With default permissions given in a workflow, I get this error, that it doesnt have enough permissions to publish a new release

https://github.com/shawnyu5/starchart/actions/runs/4249711991/jobs/7390117298

Reply:
That error makes it seem like it wants to create a new tag, which seems odd if we push our own.

Reply:
I think you're right, and this is something I didn't fully understand.  I found [this](http://practicalseries.com/1002-vcs/09-06-github.html#js--090601) helpful:

> 9.6.1
> Tags and releases—the differences
Well a tag is just a unique symbolic name attached to a particular commit point. Git supports two types of tag ([§ 6.3.4](http://practicalseries.com/1002-vcs/06-03-example.html#js--060304)) a lightweight tag (without a message) and an annotated tag (with a message). In Brackets we only have lightweight tags.
>
>GitHub can display both. In [Figure 9.82](http://practicalseries.com/1002-vcs/09-06-github.html#) above, the tags are all lightweight, I can tell this because it tells me there is no release notes for any of the tags, this release note is what Git refers to as the tag message attached to the annotated tag. If we added a release note it would become an annotated tag.
>
>A release on the other hand is a GitHub object, it is still attached to a particular commit point, but this time it allows us to specify that a particular commit has a particular release status.
>
>GitHub converts all our tags to releases when we push them up from the local repository and that is why they are listed as both releases and tags.

So GitHub does want our manually created tag, and will use it to create a new GitHub Release (tag) object, which is why it wants elevated privileges.

Reply:
ah that makes sense. Tho I don't think it needs all those permissions there, I just put them there for now so I can move on to test the rest of the action. I should remove the ones the action does not need.